.github/workflows/ci.yml

```diff
@@ -0,0 +1,140 @@
+name: CI
+
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        numfocus_nightly: [false]
+        os: ["ubuntu-latest"]
+        pyarrow: ["0.13.0", "0.14.1", "0.15.0", "0.16.0", "0.17.1", "nightly"]
+        python: ["3.6", "3.7", "3.8"]
+        include:
+          - numfocus_nightly: true
+            os: "ubuntu-latest"
+            pyarrow: "0.17.1"
+            python: "3.8"
+          - numfocus_nightly: false
+            os: "macos-latest"
+            pyarrow: "0.17.1"
+            python: "3.8"
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      # CI setup
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Conda Bootstrap
+        uses: goanpeca/setup-miniconda@v1
+        with:
+          auto-update-conda: true
+          python-version: ${{ matrix.python }}
+      - name: Conda Config
+        run: |
+          conda config --add channels conda-forge
+          conda config --set channel_priority strict
+          conda config --set always_yes yes
+          conda config --set changeps1 no
+          conda config --set pip_interop_enabled True
+
+      # Installation
+      - name: Conda Install
+        shell: bash -l {0}
+        run: conda install --file=conda-requirements.txt --file=conda-test-requirements.txt
+      - name: Conda Install Pyarrow (non-nightly)
+        shell: bash -l {0}
+        run: conda install pyarrow==${{ matrix.pyarrow }}
+        if: matrix.pyarrow != 'nightly'
+      - name: Conda Install Pyarrow (nightly)
+        shell: bash -l {0}
+        # Install both arrow-cpp and pyarrow to make sure that we have the
+        # latest nightly of both packages. It is sadly not guaranteed that the
+        # nightlies and the latest release would otherwise work together.
+        run: conda install -c arrow-nightlies arrow-cpp pyarrow
+        if: matrix.pyarrow == 'nightly'
+      - name: Pip Instal NumFOCUS nightly
+        shell: bash -l {0}
+        # NumFOCUS nightly wheels, contains numpy and pandas
+        run: env PRE_WHEELS="https://pypi.anaconda.org/scipy-wheels-nightly/simple" pip install --pre --upgrade --timeout=60 -f $PRE_WHEELS pandas numpy
+        if: matrix.numfocus_nightly
+      - name: Pip Install blacken-docs
+        shell: bash -l {0}
+        # currently not on conda
+        run: pip install --no-deps blacken-docs
+      - name: Pip Install kartothek
+        shell: bash -l {0}
+        run: pip install --no-deps -e .
+      - name: Conda Export
+        shell: bash -l {0}
+        run: conda list --export
+
+      # Linters
```

Comment:
What about running these using `pre-commit run -a`? I like the output of `pre-commit` for them as it shows the problems for all linters.

Reply:
This has multiple issues:

- it doesn't show up nicely in github actions
- it uses yet another venv system
- the venv setup might require retries, and our current setup is broken (the black hook also fixes the files, then fails, then gets retried and then passes 🤦 )

Reply:
Another change: we're now checking ALL files, not just the ones that where touched by the PR. This is esp. important for upgrades (e.g. mypy itself), config changes and changes to "unrelated" files (e.g. changes in one file might produce errors in other files importing/using them).